snapcraft/cli/_errors.py

```diff
@@ -86,18 +96,26 @@ def exception_handler(  # noqa: C901
         - a snapcraft handled error occurs, debug=False so only the
           exception message is shown
     """
-    # TODO pickle the traceback if on a manged host so the actual host can deal with it.
     exc_info = (exception_type, exception, exception_traceback)
     exit_code = 1
-    is_snapcraft_error = issubclass(exception_type, errors.SnapcraftError)
+    is_snapcraft_error = (
+        issubclass(exception_type, errors.SnapcraftError)
+        and exception_type is not ProviderExecError
+    )
     is_snapcraft_reportable_error = issubclass(
         exception_type, errors.SnapcraftReportableError
     )
     is_raven_setup = RavenClient is not None
+    is_snapcraft_host = exc_info is not None and not os.path.isfile(TRACEBACK_HOST)
```

Comment:
can you add a comment above on why this is `is_snapcraft_host`? It doesn't parse quickly as the checks seem unrelated.

Reply:
Testing `exc_info` for `None` is unnecessary since we always get here when an exception is raised. Adding a comment to explain the rest.